### ISSUE TITLE
`start` does not run the command from `pwd`

### ISSUE THREAD

**Issue by tnthung** [nushell/nushell#13127]
### Describe the bug

Say I have the following script:

```nu
cd foo/bar
start ~/bin/foo_bar.exe
```

The expected behavior should be starting the foo_bar.exe with `./foo/bar` as its working directory, but it actually starts the executable at `.`.

### How to reproduce

1. `echo "pwd out> tmp.txt" out> tmp.nu`
2. `echo "let CUR = (pwd); cd ../; start $\"($CUR)/tmp.nu\"" out> tmp2.nu`
3. `./tmp2.nu`

### Expected behavior

The `pwd` inside of `tmp.txt` should be `CUR/../`, but got `CUR`.

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.88.1                                   |
| branch             |                                          |
| commit_hash        | fd56768fdcc1acbebcb75745600fc7419f72833d |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.72.1 (d5c2e9c34 2023-09-13)      |
| rust_channel       | 1.72.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.72.1 (103a7ff2e 2023-08-15)      |
| build_time         | 2023-12-14 17:53:48 +00:00               |
| build_rust_channel | release                                  |
| allocator          | mimalloc                                 |
| features           | default, sqlite, trash, which, zip       |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
I didn't think the `start` command was meant run executables, according to the help.
```
> help start
Open a folder, file or website in the default application or viewer.
```
The crate itself says `Use this library to open a path or URL using the program configured on the system.`

**Comment by tnthung** [comment]
Yes, it's opening the file with the default application. But even then, the application should still be opened with respect to CWD, isn't it? Otherwise, the environment may not be valid for the specific app to function correctly.

**Comment by fdncred** [comment]
I'm trying to follow your instructions but `start anyfile.nu` doesn't work, so I'm not sure what this is supposed to do.

**Comment by fdncred** [comment]
I got this working
```nushell
open tmp.nu
pwd out> tmp.txt
open tmp2.nu
let CUR = (pwd);cd ../;nu $"($CUR)/tmp.nu"
nu tmp2.nu
open ../tmp.txt
shows the correct ../ path
```

**Comment by tnthung** [comment]
On my machine, `nu.exe` is set as the default app for `*.nu`. So `start tmp2.nu` is basically first opening a new terminal, then running the `nu tmp2.nu` in that terminal. After the execution, the terminal will automatically be closed. It behaved a little like `start` command in Windows `CMD`.

Both my work PC & my home PC can reproduce this by following the instruction I've given. Maybe it's Windows stuff?

**Comment by IanManske** [comment]
Probably another one of our commands that doesn't account for the emulated current working directory. Related to #12975, #12976, and #13115.
